openwisp_controller/config/admin.py

```diff
@@ -559,12 +561,21 @@ class DeviceAdmin(MultitenantAdminMixin, BaseConfigAdmin, UUIDAdmin):
         fields.insert(0, "hardware_id")
     list_select_related = ("config", "organization")
 
-    class Media(BaseConfigAdmin.Media):
+    # overriding media property to allow testing in isolation
+    # as class Media is evaluated at import time making the
+    # settings not overridable in tests
+    @property
+    def media(self):
         js = BaseConfigAdmin.Media.js + [
             f"{prefix}js/tabs.js",
             f"{prefix}js/management_ip.js",
             f"{prefix}js/relevant_templates.js",
         ]
+        css = BaseConfigAdmin.Media.css["all"]
+        if app_settings.WHOIS_CONFIGURED:
+            js.append(f"{whois_prefix}js/whois_details.js")
+            css += (f"{whois_prefix}css/whois_details.css",)
```

Comment:
do we need the `whois_` prefix again in filename as it is already present in path?

```suggestion
            js.append(f"{whois_prefix}js/details.js")
            css += (f"{whois_prefix}css/details.css",)
```